DuckDB.NET.Data/DuckDBDataReader.cs

```diff
@@ -191,17 +191,28 @@ public override string GetString(int ordinal)
 
     public override T GetFieldValue<T>(int ordinal)
     {
-        return vectorReaders[ordinal].ColumnDuckDBType switch
+        var value = vectorReaders[ordinal].ColumnDuckDBType switch
         {
-            DuckDBType.List => (T)vectorReaders[ordinal].GetList(rowsReadFromCurrentChunk - 1, (dynamic)Activator.CreateInstance<T>()),
-            DuckDBType.Enum => vectorReaders[ordinal].GetEnum<T>(rowsReadFromCurrentChunk - 1),
-            _ => (T)vectorReaders[ordinal].GetValue(rowsReadFromCurrentChunk - 1)
+            DuckDBType.List => vectorReaders[ordinal].GetList(rowsReadFromCurrentChunk - 1, Activator.CreateInstance<T?>() as dynamic),
+            DuckDBType.Enum => vectorReaders[ordinal].GetEnum<T>(rowsReadFromCurrentChunk - 1) ,
+            _ => vectorReaders[ordinal].GetValue(rowsReadFromCurrentChunk - 1)
         };
+
+        if (value is not null)
+            return (T)value;
+
+        if (default(T) is null && typeof(T).IsValueType)
```

Comment:
Will this be true for any T that isn't `Nullable<>` ?

Reply:
Humm, `if (default(T) is null && typeof(T).IsValueType)` detects `Nullable<T>` ... in that case we can return `null`

Reply:
`default` will be null for Nullable<>

Reply:
Yes but if a user goes with a nullable it's what he is expecting. You have a unit test asserting this behavior if I'm not wrong.

Reply:
I tested it locally and replacing all the if statements with `return value` still pass all tests